aviary/docs/examples/OAS_subsystem.ipynb

```diff
@@ -0,0 +1,407 @@
+    "The subsystem is based on the [OpenAeroStruct aerostructural optimization with wingbox](https://mdolab-openaerostruct.readthedocs-hosted.com/en/latest/aerostructural_wingbox_walkthrough.html) example problem.\n",
+    "\n",
+    "This example performs a sub-optimization for minimum wing weight that is then used by Aviary.\n",
+    "Another use case would be to perform a structural analysis only.\n",
+    "Structural design variables would be passed to the subsystem from Aviary and wing weight and a constraint or constraints representing the structural analysis would be passed back to Aviary for use by the top level optimization.\n",
+    "\n",
+    "## Motivation\n",
+    "\n",
+    "There may be a need for a higher fidelity tool to compute wing weight instead of relying on the empirical methods in core Aviary.\n",
+    "A structural analysis external tool is usually used because of an unusual aircraft configuration that may not be predicted my Aviary empirical weight estimation methods, but in this example case it is simply a demonstration of an external capability to compute wing weight.\n",
+    "\n",
+    "## External Dependencies\n",
+    "\n",
+    "The user must install [OpenAeroStruct](https://github.com/mdolab/OpenAeroStruct) into their Python environment using the command 'pip install openaerostruct'.\n",
+    "The user must also install the [ambiance](https://ambiance.readthedocs.io/en/latest/) package using the command 'pip install ambiance'.\n",
+    "\n",
+    "## Subsystem Details\n",
+    "\n",
+    "There are two parts to building an external subsystem -- the analysis tool and the Aviary external subsystem builder for that tool.\n",
+    "The analysis tool takes inputs and parameters from Aviary and return outputs that Aviary can use to override existing variables.\n",
+    "The subsystem builder uses the Aviary external subsystem builder template to connect the analysis tool to Aviary as either a pre-mission, mission or post-mission subsystem.\n",
+    "\n",
+    "For this case, the analysis tool will compute a wing weight in the pre-mission portion of the Aviary analysis and return its value to Aviary to override the empirical wing weight value.\n",
+    "Fuel weight is passed in from Aviary as the only input currently, but other inputs may also be passed in through the subsystem builder, [OAS_wing_weight_builder](https://github.com/OpenMDAO/om-Aviary/blob/main/aviary/external_subsystems/OAS_weight/OAS_wing_weight_builder.py), by the `promotes_inputs` parameter.\n",
+    "Other Aviary variables can also be added as additional inputs based on user needs.\n",
+    "\n",
+    "```{note}\n",
+    "Some modifications of the [OAS_wing_weight_analysis](https://github.com/OpenMDAO/om-Aviary/blob/main/aviary/external_subsystems/OAS_weight/OAS_wing_weight_analysis.py) code may be necessary to add new inputs not already defined.\n",
```

Comment:
"may be necessary" or "will be necessary"?

Reply:
changed, thanks